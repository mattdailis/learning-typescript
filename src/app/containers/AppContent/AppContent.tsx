import * as React from 'react';

import { HashRouter, NavLink, Route } from 'react-router-dom';

interface ViewType {
  name: string;
  url: string;
  component: React.ComponentType;
}

interface AppContentProps {
  views: ViewType[];
}

interface AppContentState {
  activeView: ViewType;
}

class AppContent extends React.Component<AppContentProps, AppContentState>  {

  constructor(props: any) {
    super(props);
    this.state = {
      activeView: this.props.views[0],
    };
  }

  public render() {
    const { activeView } = this.state;
    const Component = activeView.component;
    return (
      <Component />
    );
  }
}

export default AppContent;
