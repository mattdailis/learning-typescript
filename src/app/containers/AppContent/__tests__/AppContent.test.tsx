import * as React from 'react';
import * as ShallowRenderer from 'react-test-renderer/shallow';
import AppContent from '../AppContent';

const views = [{
  component: () => <div>Component</div>,
  name: 'harold',
  url: 'www',
}];

describe('<AppContent />', () => {
  test('renders', () => {
    const renderer = ShallowRenderer.createRenderer();
    renderer.render(<AppContent views={views} />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
