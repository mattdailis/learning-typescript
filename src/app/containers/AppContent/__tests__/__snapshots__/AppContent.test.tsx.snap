// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppContent /> renders 1`] = `<component />`;
